src/libraries/System.DirectoryServices.Protocols/tests/DirectoryServicesProtocolsTests.cs

```diff
@@ -74,6 +74,19 @@ public void TestUnavailableNonCriticalExtension()
             _ = (SearchResponse) connection.SendRequest(searchRequest);
             // Does not throw
         }
+        [InlineData(60)]
+        [InlineData(0)]
+        [InlineData(-60)]
+        [ConditionalTheory(nameof(IsLdapConfigurationExist))]
+        public void TestSearchWithTimeLimit(int timeLimit)
+        {
+            using LdapConnection connection = GetConnection();
+
+            var searchRequest = new SearchRequest(LdapConfiguration.Configuration.SearchDn, "(objectClass=*)", SearchScope.Subtree);
+            searchRequest.TimeLimit = TimeSpan.FromSeconds(timeLimit);
+            _ = (SearchResponse)connection.SendRequest(searchRequest);
+            // Shall succeed
```

Comment:
```suggestion
            using LdapConnection connection = GetConnection();

            var searchRequest = new SearchRequest(LdapConfiguration.Configuration.SearchDn, "(objectClass=*)", SearchScope.Subtree);
            if (timeLimit < 0)
            {
                Assert.Throws<ArgumentException>(() => searchRequest.TimeLimit = TimeSpan.FromSeconds(timeLimit));
            }
            else
            {
                searchRequest.TimeLimit = TimeSpan.FromSeconds(timeLimit);
                _ = (SearchResponse)connection.SendRequest(searchRequest);
                // Shall succeed
            }
```
The negative timeLimit will throw ArgumentException which will make the test fail. It doesn't fail in CI build because this test doesn't run by default. This change should fix that.